uint32 bus_number
uint32 cs_number